src/components/PopupWithForm.js

```diff
@@ -1,11 +1,13 @@
 import Popup from "./Popup.js";
+
 export default class PopupWithForm extends Popup {
   constructor(popupSelector, handleFormSubmit) {
     super(popupSelector);
     this._popupForm = this._popupElement.querySelector(".modal__form");
+    this._submitButton = this._popupForm.querySelector("button[type='submit']"); // Reference to the submit button
     this._handleFormSubmit = handleFormSubmit;
-    //It accepts two arguments: the popup selector and a callback function,
-    //which PopupWithForm calls when the forms submit event fires.
+    // It accepts two arguments: the popup selector and a callback function,
+    // which PopupWithForm calls when the form's submit event fires.
   }
 
   close() {
```

Comment:
You shouldn’t create the **same methods** of the parent class if you don’t change them inside and only call them with `super`. The parent methods are available in any inheritor of the class.